### DISCUSSION TITLE
Inherit Script class

### DISCUSSION THREAD

**Discussion by rizlas** [netbox-community/netbox#17040] [Help Wanted!] [answered]
Hi,
I'm currently trying to inherit Script class to create a CustomScript class that contains basic objects (inputs, functions, etc.) that I would like to reuse in other scripts. I've achieved what I needed but right now when I add a script the base class (CustomScript) is added as independent script for each script that inherit from CustomScript.

Example:

```python
"""base.py."""
from extras.scripts import BooleanVar, Script


class CustomScript(Script):

    nice_flag = BooleanVar("Activate nice flag", default=False)

    def common_function(self):
        self.log_success("You're beautiful")
```

```python
"""myscript.py"""
from scripts.base import CustomScript

class BestScript(CustomScript):
    def run(self, data, commit):
        self.common_function()
```

When i add myscript.py this is the result:

![immagine](https://github.com/user-attachments/assets/9ea53ff4-7ea7-4bf7-83ae-f2ec3f019352)

Any suggestion to avoid this behavior?

**Comment by candlerb** [comment, marked as answer]
Try using a Mixin instead of subclassing.

```
class CustomScriptMixin:
    ...

class BestScript(Script, CustomScriptMixin):
    ...
```

Plenty of examples of this in the Netbox core code - just search for the text `Mixin`.  I'm not sure if `nice_flag` would be picked up this way though, and if not, you might need to use a [subclassing hook](https://peps.python.org/pep-0487/).

**Comment by rizlas** [comment]
Seems to work, thank you!  Final example:

```python
"""base.py."""

from extras.scripts import BooleanVar


class CustomScriptMixin:

    class Meta:  # noqa: D106
        commit_default = False
        fieldsets = (
            (
                "Nice section",
                ("nice_flag",),
            ),
        )

    nice_flag = BooleanVar("Activate nice flag", default=False)

    def common_function(self):
        self.log_success("You're beautiful")
```

```python
"""myscript.py"""

from extras.scripts import Script

from scripts.base_script import CustomScriptMixin


class BestScript(Script, CustomScriptMixin):
    def run(self, data, commit):
        self.common_function()
```

**Reply by ghost** [reply]
You should make your mixin inherit from `BaseScript`.
Otherwise, your linter may complain that some functions like `log_warning` are not available.

NOTE: I expected `BaseScript` to be enough, but the child class still need to inherit `Script` as well..
